node_modules/oae-core/signin/js/signin.js

```diff
@@ -53,6 +59,33 @@ define(['jquery','oae.core'], function($, oae) {
                 // Focus into the username field
                 $('#signin-local-username', $rootel).focus();
             } else {
+                // let's add this tenant to the loggedin_tenancies cookie
+                let cookieName = 'loggedin_tenancies';
+                let loggedInTenancies = docCookies.getItem(cookieName);
+                loggedInTenancies = loggedInTenancies ? JSON.parse(loggedInTenancies) : [];
+                let currentTenant = oae.data.me.tenant;
+                let existingCookie = _.findWhere(loggedInTenancies, { alias: currentTenant.alias });
+
+                // a new cookie to be stored
+                let newCookie = (function() {
+                    return {
+                        'alias': currentTenant.alias,
+                        'displayName': currentTenant.displayName,
+                        'lastLogin': Date.now()
```

Comment:
Still using `Date.now()` here and `moment()` elsewhere - either one is fine, but we shouldn't mix the two.

Reply:
Don't understand why we can't have both, since they serve different purposes. What am I missing here?

Reply:
In this case they seem to be serving the same purpose though - logging the current time? It's mainly this line that I have problems with `eachTenancy.lastLoginFromNow = moment(parseInt(eachTenancy.lastLogin)).from(now);` - here the `eachTenancy.lastLogin` was recorded as `Date.now()` while `now` was using `moment()` - while it works fine, it just seems messy to me.

Reply:
I can have `now = new Date()` instead and use it in the `moment#Constructor` but that's probably slower (compared to `moment()`) and more verbose. I'll test that.

Reply:
Couldn't we just use `moment()` here?

Reply:
Well, `moment()` returns a `Moment` object which may be used to parse or manipulate dates but all I need is a serializable Date to store the cookie. Why would I create an object when a number suffices?

```
-> Date.now();
1513871496419 # this is what I need
-> moment();
moment("2017-12-21T15:51:40.229")
```

Reply:
Do we really need  a moment object later when getting the difference between `lastLogin` and `now` later in the code, then?

Reply:
According to the docs, the `moment#from` method accepts both a `Moment` object or a `Date` object. I had picked the less verbose and more natural way - using the former - because it seemed simpler not mixing objects (`moment.from(moment)` as opposed to `moment.from(date)`). But it's changed now.

Reply:
👍